### ISSUE TITLE
Timeline: inconsistent behaviour when changing timescale

### ISSUE THREAD

**Issue by gudleik** [acidb/mobiscroll#509]
Can be reproduced using this example:

https://demo.mobiscroll.com/jquery/timeline/daily-weekly-monthly-yearly-timeline#

Set timescale to 4-hours:
<img width="1302" alt="image" src="https://user-images.githubusercontent.com/50309/192287398-d18a1a59-7e9b-41f8-9122-c388890828cf.png">

Set timescale to 12 hours:
<img width="1302" alt="image" src="https://user-images.githubusercontent.com/50309/192287465-139ede54-5de5-49d7-b691-221cd0c3e1d2.png">

Now, set the timescale to 24 hours and then back to 12 hours. This 12-hour view is different than the previous :
<img width="1302" alt="image" src="https://user-images.githubusercontent.com/50309/192287824-77c7eab2-bbb4-490f-a4e3-f152cfa8a1a2.png">

**Comment by dioslaska** [comment]
Indeed, using inequal day splits (9:00 -> 17:00 / 12 hours) can lead to this. Do you have an actual use case which requires this setup?

**Comment by gudleik** [comment]
We have a settings panel quite similar to the one used in the demo, where the user can change timescale and display hours, like this:

<img width="711" alt="image" src="https://user-images.githubusercontent.com/50309/219315927-d045f0f6-7392-4817-8f1a-4818d38ddf88.png">
